modules/foreman_auth_source_ldap.py

```diff
@@ -0,0 +1,212 @@
+  server_url:
+    description: URL of Foreman server
+    required: true
+  username:
+    description: Username accessing Foreman server
+    required: true
+  password:
+    description: Password of user accessing Foreman server
+    required: true
+  validate_certs:
+    aliases: [ verify_ssl ]
+    description: Whether or not to verify the TLS certificates of Foreman server
+    default: true
+    type: bool
+  state:
+    description: State ot the LDAP authentication source
+    default: present
+    choices: ["present", "absent"]
+'''
+
+EXAMPLES = '''
+- name: LDAP Authentication source
+  foreman_auth_source_ldap:
+    name: "Example LDAP"
+    host: "ldap.example.org"
+    server_url: "https://foreman.example.com"
+    username: "admin"
+    password: "secret"
+    validate_certs: False
+    state: present
+
+- name: LDAP Authentication with automatic registration
+  foreman_auth_source_ldap:
+    name: "Example LDAP"
+    host: "ldap.example.org"
+    onthefly_register: True
+    account: ansible@example.com
+    account_password: secret
+    attr_login: uid
+    attr_firstname: givenName
+    attr_lastname: sn
+    attr_mail: mail
+    server_url: "https://foreman.example.com"
+    username: "admin"
+    password: "secret"
+    validate_certs: False
+    state: present
+'''
+
+RETURN = ''' # '''
+
+from ansible.module_utils.foreman_helper import (
+    sanitize_entity_dict,
+    ForemanEntityApypieAnsibleModule,
+)
+
+# This is the only true source for names (and conversions thereof)
+name_map = {
+    'name': 'name',
+    'host': 'host',
+    'port': 'port',
+    'account': 'account',
+    'account_password': 'account_password',
+    'base_dn': 'base_dn',
+    'attr_login': 'attr_login',
+    'attr_firstname': 'attr_firstname',
+    'attr_lastname': 'attr_lastname',
+    'attr_mail': 'attr_mail',
+    'attr_photo': 'attr_photo',
+    'onthefly_register': 'onthefly_register',
+    'usergroup_sync': 'usergroup_sync',
+    'tls': 'tls',
+    'groups_base': 'groups_base',
+    'server_type': 'server_type',
+    'ldap_filter': 'ldap_filter',
+}
+
+
+def main():
+    module = ForemanEntityApypieAnsibleModule(
+        argument_spec=dict(
+            name=dict(required=True),
+            host=dict(required=True),
+            port=dict(type='int'),
+            account=dict(),
+            account_password=dict(no_log=True),
+            base_dn=dict(),
+            attr_login=dict(),
+            attr_firstname=dict(),
+            attr_lastname=dict(),
+            attr_mail=dict(),
+            attr_photo=dict(),
+            onthefly_register=dict(type='bool'),
+            usergroup_sync=dict(type='bool'),
+            tls=dict(type='bool'),
+            groups_base=dict(),
+            server_type=dict(choices=["free_ipa", "active_directory", "posix"]),
+            ldap_filter=dict(),
+        ),
+        supports_check_mode=True,
+    )
+
+    (entity_dict, state) = module.parse_params()
+
+    module.connect()
+
+    try:
```

Comment:
given #281 is merged, we can drop the try/except here

Reply:
Done!